p2p/src/net/mock/peer.rs

```diff
@@ -74,10 +82,25 @@ where
         p2p_config: Arc<P2pConfig>,
         socket: T::Stream,
         receiver_address: Option<PeerAddress>,
+        handshake_nonce_store: Arc<HandshakeNonceStore>,
         tx: mpsc::Sender<(MockPeerId, PeerEvent)>,
         rx: mpsc::Receiver<MockEvent>,
     ) -> Self {
         let socket = BufferedTranscoder::new(socket);
+
+        // Generate random nonce
+        let handshake_nonce = match role {
+            Role::Inbound => None,
+            Role::Outbound => Some(make_true_rng().gen()),
+        };
+
+        if let Some(handshake_nonce) = handshake_nonce {
+            handshake_nonce_store
+                .lock()
+                .expect("handshake_nonce_store mutex must be valid")
+                .insert(handshake_nonce);
+        }
```

Comment:
Minor: I'd probably prefer to fold the `if let` into the `Role::Outbound` arm of the preceding `match`. That way you don't need to analyse data flow to see what control flow paths are possible.

Reply:
Sure, updated.